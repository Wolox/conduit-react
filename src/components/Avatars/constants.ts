import Man1 from 'assets/avatars/Man1.png';
import Man2 from 'assets/avatars/Man2.png';
import Man3 from 'assets/avatars/Man3.png';
import Man4 from 'assets/avatars/Man4.png';
import Woman1 from 'assets/avatars/Woman1.png';
import Woman2 from 'assets/avatars/Woman2.png';
import Woman3 from 'assets/avatars/Woman3.png';
import Woman4 from 'assets/avatars/Woman4.png';
import { Avatar } from 'types/Avatar';

// export const AVATARS = [Man2, Man3, Man4, Woman1, Woman2, Woman3, Woman4];

export const AVATARS: Avatar[] = [
  { name: 'Man1', icon: Man1 },
  { name: 'Man2', icon: Man2 },
  { name: 'Man3', icon: Man3 },
  { name: 'Man4', icon: Man4 },
  { name: 'Woman1', icon: Woman1 },
  { name: 'Woman2', icon: Woman2 },
  { name: 'Woman3', icon: Woman3 },
  { name: 'Woman4', icon: Woman4 }
];
