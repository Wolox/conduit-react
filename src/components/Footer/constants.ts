export const urlWolox = 'https://www.wolox.com.ar/';
