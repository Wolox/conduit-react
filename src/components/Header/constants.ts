import { faEdit, faCog, faUser, faSignInAlt, faUserPlus } from '@fortawesome/free-solid-svg-icons';

import PATHS from 'components/Routes/paths';

const itemsMenu = [
  {
    text: 'Header:signIn',
    href: PATHS.login,
    isProtected: false,
    icon: faSignInAlt
  },
  {
    text: 'Header:signUp',
    href: PATHS.register,
    isProtected: false,
    icon: faUserPlus
  },
  {
    text: 'Header:profile',
    href: PATHS.profile,
    isProtected: true,
    icon: faUser
  },
  {
    text: 'Header:editor',
    href: PATHS.editor,
    isProtected: true,
    icon: faEdit
  },
  {
    text: 'Header:settings',
    href: PATHS.settings,
    isProtected: true,
    icon: faCog
  }
];

export default itemsMenu;
