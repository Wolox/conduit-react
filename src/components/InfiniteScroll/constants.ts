export const THRESHOLD = 0.9;
