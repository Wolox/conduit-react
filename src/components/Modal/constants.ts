export const KEY_ESCAPE = 'Escape';
export const TEST_ID_SHADOW = 'shadow';
