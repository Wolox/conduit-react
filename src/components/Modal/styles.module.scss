.content-modal {
  align-items: center;
  background: rgba(0, 0, 0, 0.5);
  bottom: 0;
  display: flex;
  justify-content: center;
  left: 0;
  overflow: hidden;
  position: fixed;
  right: 0;
  top: 0;
  transition: all 0.5;
  z-index: 1;
}

.hidden {
  display: none;
}
