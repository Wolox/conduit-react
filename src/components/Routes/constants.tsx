/* eslint-disable @typescript-eslint/naming-convention */
import { lazy } from 'react';

import { UserProfile } from 'types/User';

import PATHS from './paths';

const Home = lazy(() => import('screens/Dashboard'));
const EditorScreen = lazy(() => import('screens/EditorScreen'));
const Login = lazy(() => import('screens/Authentication/Login'));
const Register = lazy(() => import('screens/Authentication/Register'));
const Article = lazy(() => import('screens/Article'));
const Profile = lazy(() => import('screens/Profile'));
const Error = lazy(() => import('screens/Error'));
const Settings = lazy(() => import('screens/Settings'));
// Add imports for screens above (FOR GENERATORS, DO NOT REMOVE)

const MAIN_PUBLIC_PATH = PATHS.login;
const MAIN_PRIVATE_PATH = PATHS.home;

/* When adding routes, add them ABOVE the Home route
 * or it will redirect you to incorrect screens */
export const ROUTES = [
  // Leaving this as an example for then the Login screen exists
  {
    exact: true,
    path: PATHS.login,
    component: Login,
    title: 'Routes:loginTitle',
    description: 'Routes:loginDescription',
    redirectTo: (user: UserProfile | null) => (user ? MAIN_PRIVATE_PATH : undefined)
  },
  {
    exact: true,
    path: PATHS.register,
    component: Register,
    title: 'Routes:registerTitle',
    redirectTo: (user: UserProfile | null) => (user ? MAIN_PRIVATE_PATH : undefined)
  },
  {
    exact: true,
    path: PATHS.editor,
    component: EditorScreen,
    title: 'Routes:editorTitle',
    redirectTo: (user: UserProfile | null) => (user ? undefined : MAIN_PUBLIC_PATH)
  },
  {
    exact: false,
    path: PATHS.editorBySlug,
    component: EditorScreen,
    title: 'Routes:editorTitle',
    redirectTo: (user: UserProfile | null) => (user ? undefined : MAIN_PUBLIC_PATH)
  },
  {
    exact: true,
    path: PATHS.errorScreen,
    component: Error,
    title: 'Routes:error',
    redirectTo: (user: UserProfile | null) => (user ? undefined : MAIN_PUBLIC_PATH)
  },
  {
    path: PATHS.article,
    component: Article,
    title: 'Routes:articleTitle'
  },
  {
    exact: true,
    path: PATHS.profile,
    component: Profile,
    title: 'Routes:profileTitle'
  },
  {
    exact: true,
    path: PATHS.settings,
    component: Settings,
    title: 'Routes:settings',
    redirectTo: (user: UserProfile | null) => (user ? undefined : MAIN_PUBLIC_PATH)
  },
  {
    exact: false,
    path: PATHS.home,
    component: Home,
    title: 'Routes:homeTitle',
    description: 'Routes:homeDescription'
  }
];
