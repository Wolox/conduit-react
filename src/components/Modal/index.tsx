import cn from 'classnames';
import React, { useEffect, useRef } from 'react';

import { KEY_ESCAPE, TEST_ID_CONTENT, TEST_ID_SHADOW } from './constants';
import styles from './styles.module.scss';

interface Props {
  children: React.ReactNode;
  closeModal: () => void;
  open: boolean;
  className?: string;
}

function Modal({ children, closeModal, open, className }: Props) {
  const contentModal = useRef(null);

  const handleOnClick = (e: React.MouseEvent<HTMLDivElement>) => {
    if (e.target === contentModal.current) {
      closeModal();
    }
  };

  useEffect(() => {
    const close = (e: KeyboardEvent) => {
      if (e.key === KEY_ESCAPE) {
        closeModal();
      }
    };
    window.addEventListener('keydown', close);
    return () => window.removeEventListener('keydown', close);
  }, [closeModal]);
  return (
    <>
      {open && (
        <div
          className={styles.contentModal}
          onClick={handleOnClick}
          ref={contentModal}
          data-testid={TEST_ID_SHADOW}
        >
          <div className={cn(className, styles.modalBody)} data-testid={TEST_ID_CONTENT}>
            {children}
          </div>
        </div>
      )}
    </>
  );
}

export default Modal;
