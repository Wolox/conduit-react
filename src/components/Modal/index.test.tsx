import { createMemoryHistory } from 'history';
import { ReactNode } from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Router } from 'react-router';

import { TEST_ID_CONTENT, TEST_ID_SHADOW } from './constants';

import Modal from '.';

function Wrapped({ children }: { children: ReactNode }) {
  const queryClient = new QueryClient();
  const history = createMemoryHistory();
  return (
    <QueryClientProvider client={queryClient}>
      <Router history={history}>{children}</Router>
    </QueryClientProvider>
  );
}

describe('Component Modal', () => {
  test('Renders component', () => {
    const { asFragment } = render(
      <Wrapped>
        <Modal closeModal={jest.fn()} open>
          <div>Hello Word</div>
        </Modal>
      </Wrapped>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('Display the content children in the modal', () => {
    render(
      <Wrapped>
        <Modal closeModal={jest.fn()} open>
          <div>Hello Word</div>
        </Modal>
      </Wrapped>
    );
    const contentModal = screen.getByText('Hello Word');
    expect(contentModal).toBeInTheDocument();
  });

  test('Not Display the modal when props open is false', () => {
    render(
      <Wrapped>
        <Modal closeModal={jest.fn()} open={false}>
          <div>Hello Word</div>
        </Modal>
      </Wrapped>
    );
    const contentModal = screen.queryByText('Hello Word');
    expect(contentModal).not.toBeInTheDocument();
  });

  test('Function for close modal is called when press Escape', () => {
    const closeModal = jest.fn();
    render(
      <Wrapped>
        <Modal closeModal={closeModal} open>
          <div>Hello Word</div>
        </Modal>
      </Wrapped>
    );

    const contentModal = screen.getByText('Hello Word');
    fireEvent.keyDown(contentModal, {
      key: 'Escape',
      code: 'Escape',
      keyCode: 27,
      charCode: 27
    });

    expect(closeModal).toBeCalled();
  });

  test('The function to close modal is called when clicking outside the modal is pressed', () => {
    const closeModal = jest.fn();
    render(
      <Wrapped>
        <Modal closeModal={closeModal} open>
          <div>Hello Word</div>
        </Modal>
      </Wrapped>
    );

    const contentModal = screen.getByTestId(TEST_ID_SHADOW);
    userEvent.click(contentModal);

    expect(closeModal).toBeCalled();
  });

  test('Add the class in the container of the children when it is sent in the props', () => {
    const classSend = 'classSend';
    render(
      <Wrapped>
        <Modal closeModal={jest.fn()} open className={classSend}>
          <div>Hello Word</div>
        </Modal>
      </Wrapped>
    );
    const contentChildren = screen.getByTestId(TEST_ID_CONTENT);
    expect(contentChildren).toHaveClass(classSend);
  });
});
