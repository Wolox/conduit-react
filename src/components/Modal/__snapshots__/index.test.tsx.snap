// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Modal Renders component 1`] = `
<DocumentFragment>
  <div
    class="contentModal"
    data-testid="shadow"
  >
    <div
      class="modalBody"
      data-testid="contentChildren"
    >
      <div>
        Hello Word
      </div>
    </div>
  </div>
</DocumentFragment>
`;
